config/config.yml

```diff
@@ -503,3 +503,7 @@ STAT_COUNTER_BATCH_SIZE: 100
 # Number of hours to retain the original creator for works that have been
 # orphaned.
 ORIGINAL_CREATOR_TTL_HOURS: 72
+
+# Regex for the site's domains, used to validate that user-supplied
+# links are located on the Archive when appropriate.
+APP_URL_REGEX: "^(https?:\\/\\/)?(www\\.|(insecure\\.))?(archiveofourown|ao3)\\.(org|com).*"
```

Comment:
I'm not sure about having a regex in the config file. I think configs should be simple values. Regexes are complicated and you won't know they don't work until run time, so it's easy for us to mistype it when overriding in local.yml and not notice because it doesn't go through our usual testing process.

Following the discussion in [AO3-6496](https://otwarchive.atlassian.net/browse/AO3-6496), I think we can use `Addressable::URI.heuristic_parse` (which handles URLs without schemes for us, see test cases in #4483). Once parsed, we can match `.host` against the existing config value `PERMITTED_HOSTS`.

Reply:
Oh that's a neat method! I agree, that's a much cleaner way to do things. Would we want systems to override `PERMITTED_HOSTS` in production to avoid letting staging URLs through there?

Reply:
Yeah, we should have `PERMITTED_HOSTS` be different between staging and production in that case.

Reply:
I've pulled out the staging values, since that way would impact fewer people if the override was delayed for some reason

Reply:
You can leave the staging values in, for public tracking purposes. Currently, `PERMITTED_HOSTS` is always overridden by our deploy process for staging/production, so it needs to be handled before we merge regardless.